src/features/ProfileView/Profile.History.js

```diff
@@ -44,26 +41,56 @@ const ProfileHistory = () => {
     {restauraunt: 'Taqueria El Rey', hostStatus: 'Host', bill: 8.49}
   ];
 
+  const [biteHistory, setBiteHistory] = useState(mockBites);
+  const { state: { userId }, dispatch } = useContext(BiteShareContext);
+
   const renderBite = ({item, index}) => (<PreviousBite meal={item} index={index}/>);
 
-  //TODO
-  //Query previous bites from db
-  //Restauraunt Name - Guest / Host Status - Price
-  //Place in array in state to be rendered via Flatlist
+  useEffect(() => {
+    getAllDocuments(`users/${userId}/transactions`)
+      .then((user) => {
+        let cleanedTransactions = biteHistory;
+
+        user.forEach((session) => {
+          readASingleDocument(`users/${userId}/transactions`, session.id)
+            .then((summary) => {
+              summary = summary.data();
+              let bite = {
+                restauraunt: summary.restaurantName,
+                bill: summary.individualBills,
+                hostStatus: summary.role || 'guest'
+              };
+              cleanedTransactions.push(bite);
+            })
+            .catch((err) => {
+              console.log('Error pulling transaction history for meal session: ', session.id);
+            });
+        });
+
+        setBiteHistory(cleanedTransactions);
+      })
```

Comment:
```suggestion
      })
      .then(() => {
        setBiteHistory(cleanedTransactions);
      })
```

to make sure that it sets the state only after the most recent transaction is pulled

Reply:
lol to complete we also need to move `let cleanedTransactions = biteHistory;` to line 50 otherwise the `.then` block won't have access to `cleanedTransactions`